front/src/modules/rollingStock/helpers/utils.ts

```diff
@@ -375,3 +469,86 @@ export const splitRollingStockProperties = (
   });
   return groupBy(displayedProperties, 'side') as { [key: string]: SchemaProperty[] };
 };
+
+/**
+ * According the conversion factors schema, convert a value if
+ * the factors exist in the schema
+ */
+export const convertUnits = (prevUnit: string, newUnit: string, prevValue: number): number => {
+  if (prevUnit in conversionFactorsSchema && newUnit in conversionFactorsSchema[prevUnit]) {
+    return conversionFactorsSchema[prevUnit][newUnit] * prevValue;
+  }
+  return prevValue;
+};
+
+/**
+ * For the rollingstock resistance (a, b or c), check if its unit
+ * to be converted or the unit to convert to is by ton (i.e. daN/t).
+ *
+ * As the resistance mass unit can only be in ton, return the converted
+ * resistance depending if the rollingstock's current mass is in kg or ton
+ */
+export const handleConverWithMass = (
+  previousUnit: string,
+  newUnit: string,
+  currentMassValue: number,
+  currentMassUnit: string,
+  previousValue: number
+) => {
+  let convertedValue = previousValue;
+  if (newUnit.slice(-1) === 't') {
+    const formatedNewUnit = newUnit.slice(0, -2);
+    convertedValue = convertUnits(previousUnit, formatedNewUnit, convertedValue);
+    if (currentMassUnit === 'kg') {
+      currentMassValue = convertUnits('kg', 't', currentMassValue);
+    }
+    convertedValue /= currentMassValue;
+  }
+  if (previousUnit.slice(-1) === 't') {
+    const formatedPreviousUnit = previousUnit.slice(0, -2);
+    convertedValue = convertUnits(formatedPreviousUnit, newUnit, convertedValue);
+    if (currentMassUnit === 'kg') {
+      currentMassValue = convertUnits('kg', 't', currentMassValue);
+    }
+    convertedValue *= currentMassValue;
+  }
+  return convertedValue;
+};
```

Comment:
- Rename the function to convertUnitsWithMass for consistency with convertUnits.
- Use endsWith to check unit suffixes, enhancing code readability.
- Calculate the mass in tons only once for cleaner logic.
Not sure if for the part of the function that checks units (`if (newUnit.endsWith('t'))` and `if (previousUnit.endsWith('t'))` ), we should use two separate ifs or go with an if-else-if structure. 
```suggestion
export const convertUnitWithMass = (
  previousUnit: string,
  newUnit: string,
  currentMassValue: number,
  currentMassUnit: string,
  previousValue: number
) => {
  let convertedValue = previousValue;

  const massInTons =
    currentMassUnit === 'kg' ? convertUnits('kg', 't', currentMassValue) : currentMassValue;
  if (newUnit.endsWith('t')) {
    convertedValue = convertUnits(previousUnit, newUnit.slice(0, -2), previousValue) / massInTons;
  }

  if (previousUnit.endsWith('t')) {
    convertedValue = convertUnits(previousUnit.slice(0, -2), newUnit, previousValue) * massInTons;
  }
  return convertedValue;
};
```

Reply:
I used twice the kg check because I didn't want to to convert the mass if it was not needed (not any units ending by 't'), but you're right, its way cleaner like this and it's not a big computation.

Renamed and switch with endsWith().